src/client/feature_snapshot.rs

```diff
@@ -118,17 +117,6 @@ impl FeatureSnapshot {
             Ok(self.disabled_value.clone())
         }
     }
-
-    fn send_metering(&self, entity_id: &str, segment_id: Option<&str>) {
-        if let Some(metering) = self.metering.as_ref() {
-            if let Err(e) = metering.record_feature_evaluation(&self.name, entity_id, segment_id) {
-                warn!(
-                    "Fail to enqueue metering data for feature '{}': {e}",
-                    self.name
-                );
-            }
-        }
-    }
```

Comment:
This logic will be owned by the `metering` module. The `Feature/PropertySnapshot` will just implement a `MeteringSubject` trait that provides a `self.record_evaluation`.